ntbs-service-unit-tests/DataMigration/NotificationMapperTest.cs

```diff
@@ -454,6 +453,36 @@ public async Task correctlyMaps_DuplicateTravelCountries()
             Assert.Null(notification.TravelDetails.StayLengthInMonths3);
         }
 
+        // Data for this has been based on real regression examples, but with care taken to anonymize it
+        // This is based on NTBS-2388
+        [Fact]
+        public async Task correctlyMaps_DenotifiedStatusIfOtherwiseShouldBeClosed()
+        {
+            // Arrange
+            const int runId = 12345;
+            var legacyIds = new List<string> { "249398" };
+            SetupNotificationsInGroups(("249398", "10"));
+
+            const string malvernCode = "TBS0656";
+            _hospitalToTbServiceCodeDict = new Dictionary<Guid, TBService>
+            {
+                {new Guid("33464912-E5B1-4998-AFCA-083C3AE65A80"), new TBService {Code = malvernCode}},
+            };
+
+            // Act
+            var notification = (await _notificationMapper.GetNotificationsGroupedByPatient(null,
+                    runId,
+                    legacyIds))
+                .SelectMany(group => group)
+                .Single();
+
+            // Assert
+            Assert.Equal(NotificationStatus.Denotified ,notification.NotificationStatus);
+            // Change status to see if notification otherwise would have been closed
+            notification.NotificationStatus = NotificationStatus.Notified;
+            Assert.True(notification.ShouldBeClosed());
```

Comment:
This is a slightly strange thing to do - we're essentially verifying that the test data has been set up correctly... which is really something that we should be able to rely on. I think we should get rid of it.